dallinger/frontend/static/scripts/dallinger2.js

```diff
@@ -632,9 +633,10 @@ var dallinger = (function () {
       number: 1,
       response: JSON.stringify(formDict)
     });
-    spinner.freeze([$('form :input')]);
-    xhr.done(dlgr.submitAssignment);
-    xhr.always(function () { spinner.unfreeze(); });
+    spinner.freeze([$inputs, $button]);
+    xhr.done(function () {
+      dlgr.submitAssignment().always(function () { spinner.unfreeze(); });
```

Comment:
Won't this prevent `unfreeze` if the POST succeeds, but the `submitAssignment` fails? Is that desirable?

Reply:
I don't think so, because the unfreeze is in an `always()` call on the return of `submitAssignment()`, but it _will_ prevent unfreeze if the first POST fails.  The problem is we don't want to unfreeze before the `submitAssignment` call finishes.

Reply:
Yeah, the POST fails so `submitAssignment()` call never happens is the case I was thinking about when I decided to comment and then I wrote the opposite for some reason. So this is a change in behavior. I guess if the POST fails we expect the normal error handling process and don't worry about `unfreeze`?

Reply:
This is thoroughly tested and working correctly now (revisions made).

Reply:
I couldn't stand looking at the crappy error forms any more so tidied those up a little bit also.